lib/matplotlib/patches.py

```diff
@@ -1209,21 +1209,15 @@ def _recompute_path(self):
             # followed by a reversed and scaled inner ring
             v1 = arc.vertices
             v2 = arc.vertices[::-1] * (self.r - self.width) / self.r
-            v = np.concatenate([v1, v2, [v1[0, :], (0, 0)]])
-            c = np.concatenate([
-                arc.codes, arc.codes, [connector, Path.CLOSEPOLY]])
-            c[len(arc.codes)] = connector
+            v = np.concatenate([v1, v2, [(0, 0)]])
+            c = [*arc.codes, connector, *arc.codes[1:], Path.CLOSEPOLY]
```

Comment:
Maybe not significant, but creating the list costs a little extra time and memory. Path turns it into an array afterwards.

itertools.chain would be a bit cheaper. Not sure whether it’s worth it though.

Reply:
But previously you also had to create the list just to pass it as argument to np.concatenate; using itertools.chain doesn't really help either because np.array() (which gets called somewhere, whether here or in the Path constructor) doesn't take generators as input so you again need to convert to a list first.

The only thing that can be a bit more costly is the explicit iteration over arc.codes (`*arc.codes`), but these are quite short iterators anyways.

Reply:
You’re right.